component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java

```diff
@@ -613,7 +613,21 @@ private void storeFile(Identity streamOwner, ExoSocialActivity activity) throws
             LOG.warn("No file storage plugin for datasource with name '{}'", activityFile.getStorage());
           }
         }
-        activityFileStoragePlugin.storeAttachments(activity, streamOwner, files.toArray(new ActivityFile[files.size()]));
+        if (StringUtils.isEmpty(activityFile.getId()) && StringUtils.isNotEmpty(activityFile.getUploadId())) {
+          try {
+            activityFileStoragePlugin.storeAttachments(activity, streamOwner, activityFile);
+          } catch (Exception e) {
+            LOG.error("Error while adding attachment to activity " + activityFile.getId(), e);
```

Comment:
activityFile.getId() is always empty in this case.

Reply:
Changed to `getName()`